plugins/inputs/oomkiller/oomkiller.go

```diff
@@ -0,0 +1,71 @@
+package oomkiller
+
+import (
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/tail"
+)
+
+type Oomkiller struct {
+	Logfile     string `toml:"logfile"`
+}
+
+var sampleConfig = `
+  ## Logfile where oom killer is reflected
+  logfile = "/var/log/messages"
+`
+
+// SampleConfig returns a sample configuration block
+func (a *Oomkiller) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description just returns a short description of the Mesos plugin
+func (a *Oomkiller) Description() string {
+	return "Telegraf plugin for gathering metrics from kernel Oomkiller"
+}
+
+func (a *Oomkiller) SetDefaults() {
+	if a.Logfile == "" {
+		log.Println("I! [oomkiller] Missing logfile value, setting default value (/var/log/messages)")
+		a.Logfile = "/var/log/messages"
+	}
+}
+
+func IsOomkillerEvent(line string) bool {
+	matched, _ := regexp.MatchString("invoked oom-killer", line)
+	return matched
+}
+
+// Gather() metrics from Oomkiller
+func (a *Oomkiller) Gather(acc telegraf.Accumulator) error {
+	a.SetDefaults()
+	//This is to NOT start tailing log from the beginning
+	var seek *tail.SeekInfo
+	seek = &tail.SeekInfo{ Whence: 2, Offset: 0, }
+	t, _ := tail.TailFile(a.Logfile, tail.Config{ Follow: true, Location:  seek, ReOpen: true})
```

Comment:
tail.TailFile returns the pointer and error. Is there any reason to discard the error?

Reply:
Fixed on https://github.com/medallia/telegraf/pull/16/commits/6218653ca65f5d0c59e755d8d78a7a5385710bfe